### DISCUSSION TITLE
Feature Request: More granular over LanguageTool languages

### DISCUSSION THREAD

**Discussion by lesydimitri** [Syslifters/sysreptor#216] [General]
Hi!

I just started playing around with the Pro version and I very much like the LanguageTool integration. However, it would be great to have more control over the languages. More specifically, I would like to be able to select language "variants" that LanguageTool offers.

For example, for English clients, I would prefer to write in 'en-GB' instead of 'en-US'.

This probably would also make sense for other languages (for example 'de-DE' vs 'de-AT', Jänner/Januar, ...).

Kind regards
Dimitri Lesy

**Comment by lesydimitri** [comment]
As far as I can see from the code, supported languages are hardcoded here:

https://github.com/Syslifters/sysreptor/blob/0db036362d9bb0f3f31e9e3d753e491f8f9abc1a/api/src/reportcreator_api/pentests/models/common.py#L123C1-L153


I tried changing this to have "British English" and "American English" and that all seems to work as expected:
![image](https://github.com/Syslifters/sysreptor/assets/1854585/30863046-026c-4d72-9a48-195375abdd00)
![image](https://github.com/Syslifters/sysreptor/assets/1854585/79a71118-a317-4f1b-b5c1-28dfa5858d7b)


Would it be possible to extend this list to all language codes supported by LanguageTool or to hand over control to the administrator, either in the web app or via the configuration file?

Kind regards
Dimitri Lesy

**Comment by aronmolnar** [comment]
Yes, we can do this.  
I would, however, not add the dialects to the default languages, but they should rather have to be configured via the `PREFERRED_LANGUAGES` variables.

What do you think @MWedl?

**Reply by MWedl** [reply]
I would prefer adding language variants to the list of language selections. Configuring language variants via an environment variable affects only a single instance (where it is configured). The language variant cannot move to other instances. 

For example, when projects are exported and imported on other instances with different configured language variants the project might be rendered differently. The language variant is passed to HTML and used in PDF rendering for formatting dates. Different language variants might use different date formats. Additonally, projects would be filled with a lot of spellcheck errors (e.g. "color" vs "colour").

When we instead extend the language list, the language variant moves together with the project. This ensures that projects are rendered the same on all instances.

**Comment by MWedl** [comment]
Added language variants in https://github.com/Syslifters/sysreptor/releases/tag/2024.19
